src/Microsoft.SqlTools.ServiceLayer/LanguageServices/LanguageService.cs

```diff
@@ -1712,6 +1712,11 @@ internal string ParseStatementAtPosition(string sql, int line, int column)
                             if (statement.StartLocation.LineNumber <= parserLine 
                                 && statement.EndLocation.LineNumber >= parserLine)
                             {
+                                if (statement.EndLocation.LineNumber == parserLine && statement.EndLocation.ColumnNumber < parserColumn
```

Comment:
can also verify this works for lines such as "SELECT 1;   " with the cursor in the blank space following the ";" character?  I expect users would expect the command to work if they are on a line with a single statement, but outside what the parser strictly considers the statement.  If there are multiple statements on the line and we can determine which is the "active statement" then we should just continue.

Reply:
Added logic to always execute the single statement on a line if there is only one

Reply:
thanks!  looks good to me.